### PR TITLE
analysis_options: move to language - strict-casts:true

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

**Review by jakemac53 — APPROVED** [review]
We should also bump the versions to dev versions

File: pkgs/test_api/lib/src/backend/metadata.dart [hunk 1]
```diff
@@ -90,7 +90,7 @@ class Metadata {
 
       Timeout? timeout;
       dynamic skip;
-      for (var metadatum in metadata) {
+      for (var metadatum in metadata as List) {
```
[the remark this answers is not in the document]
Comment (kevmoo) [reply pkgs/test_api/lib/src/backend/metadata.dart:93]: I'll look...

File: pkgs/test_api/lib/src/backend/metadata.dart [hunk 2]
```diff
@@ -263,7 +263,7 @@ class Metadata {
         _retry = serialized['retry'] as int?,
         tags = Set.from(serialized['tags'] as Iterable),
         onPlatform = {
-          for (var pair in serialized['onPlatform'])
+          for (var pair in serialized['onPlatform'] as List)
```
[the remark this answers is not in the document]
Comment (kevmoo) [reply pkgs/test_api/lib/src/backend/metadata.dart:266]: hrm...okay

File: pkgs/test_api/lib/src/backend/metadata.dart [hunk 3]
```diff
@@ -81,37 +81,37 @@ class Metadata {
...
-            throw ArgumentError('Only a single Timeout may be declared for '
-                '"$platform".');
-          }
-
-          timeout = metadatum;
-        } else if (metadatum is Skip) {
-          if (skip != null) {
-            throw ArgumentError('Only a single Skip may be declared for '
-                '"$platform".');
+      } else if (metadata is List) {
```
Comment (natebosch) [inline pkgs/test_api/lib/src/backend/metadata.dart:84]: This is a behavior change. When the argument was a `Timeout` it is now effectively ignored I think.

I think this is more readable with an extra helper method.

```dart
  /// Parses a user-provided map into the value for [onPlatform].
  static Map<PlatformSelector, Metadata> _parseOnPlatform(
      Map<String, dynamic>? onPlatform) {
    if (onPlatform == null) return {};

    var result = <PlatformSelector, Metadata>{};
    onPlatform.forEach((platform, metadata) {
      var selector = PlatformSelector.parse(platform);
      if (metadata is Timeout || metadata is Skip) {
        result[selector] = _parsePlatformOptions(platform, [metadata]);
      } else if (metadata is List) {
        result[selector] = _parsePlatformOptions(platform, metadata);
      }
      throw ArgumentError('Metadata for platform "$platform" must be a '
          'Timeout, Skip, or List of those; was "$metadata".');
    });
    return result;
  }

  static Metadata _parsePlatformOptions(
      String platform, List<dynamic> metadata) {
    Timeout? timeout;
    dynamic skip;
    for (var metadatum in metadata) {
      if (metadatum is Timeout) {
        if (timeout != null) {
          throw ArgumentError('Only a single Timeout may be declared for '
              '"$platform".');
        }

        timeout = metadatum;
      } else if (metadatum is Skip) {
        if (skip != null) {
          throw ArgumentError('Only a single Skip may be declared for '
              '"$platform".');
        }

        skip = metadatum.reason ?? true;
      } else {
        throw ArgumentError('Metadata for platform "$platform" must be a '
            'Timeout, Skip, or List of those; was "$metadata".');
      }
    }

    return Metadata.parse(timeout: timeout, skip: skip);
  }
```

**Review by natebosch — APPROVED** [review]

**Comment by kevmoo** [comment]
@natebosch @jakemac53 – look at the failures. I'm at a loss...

**Comment by natebosch** [comment]
> look at the failures. I'm at a loss...

I'm looking... I'm also confused so far

**Comment by natebosch** [comment]
I missed an else, will check any other tests that fail

File: pkgs/test_api/lib/src/backend/metadata.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (kevmoo) [reply pkgs/test_api/lib/src/backend/metadata.dart:270]: it's not really a dynamic invocation. We're just using a `List` like something more typed.

[the remark this answers is not in the document]
Comment (natebosch) [reply pkgs/test_api/lib/src/backend/metadata.dart:270]: The `.first` and `.last` calls below are on dynamic

[the remark this answers is not in the document]
Comment (kevmoo) [reply pkgs/test_api/lib/src/backend/metadata.dart:270]: ...right

**Pull request merged**